oauth/context_helper.go

```diff
@@ -0,0 +1,32 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oauth
+
+import "context"
+
+type contextKey string
+
+func (c contextKey) String() string {
+	return string(c)
+}
+
+var (
+	NamespaceContextKey = contextKey("namespace")
```

Comment:
I would mimic what the `log` package is doing here and provide both `NamespaceFromContext` and `NamespaceIntoContext` functions. This would make it possible to keep the key private and be just an opaque type. 

I think `type contextKey struct {}` and `var namespaceContextKey = contextKey{}` would work fine without it needing to have any concrete value. I would also maybe rename it to something more concrete like `namespaceContextKeyType` just to prevent name collisions if we add some other context key in the future.